soh/soh/Enhancements/randomizer/3drando/fill.cpp

```diff
@@ -548,13 +548,13 @@ static void CalculateWotH() {
 static void FastFill(std::vector<uint32_t> items, std::vector<uint32_t> locations, bool endOnItemsEmpty = false) {
   //Loop until locations are empty, or also end if items are empty and the parameters specify to end then
   while (!locations.empty() && (!endOnItemsEmpty || !items.empty())) {
-    uint32_t loc = RandomElement(locations, true);
-    Location(loc)->SetAsHintable();
-    PlaceItemInLocation(loc, RandomElement(items, true));
-
     if (items.empty() && !endOnItemsEmpty) {
       items.push_back(GetJunkItem());
     }
+
+    uint32_t loc = RandomElement(locations, true);
+    Location(loc)->SetAsHintable();
+    PlaceItemInLocation(loc, RandomElement(items, true));
   }
```

Comment:
is this a change that needs to be PR'd to 3drando or is it a fix from 3drando that we didn't have yet?

Reply:
This chance is not in 3drando, and could be upstreamed